### PR TITLE
[OPPRO-104] Support scalar subquery expression

### PR DESCRIPTION
**Pull request by rui-mo**
## What changes were proposed in this pull request?

Supported sending the result of scalar subquery through Substrait plan.

## How was this patch tested?

locally tested

### CODE REVIEW

File: jvm/src/main/java/io/glutenproject/substrait/expression/ExpressionBuilder.java [hunk 1]
```diff
@@ -78,6 +84,43 @@ public static StringListNode makeStringList(ArrayList<String> strConstants) {
...
+            if (obj == null) {
+                return makeNullLiteral(TypeBuilder.makeDate(nullable));
+            } else {
+                return makeDateLiteral((Integer) obj);
+            }
+        } else if (dataType instanceof StringType) {
+            if (obj == null) {
+                return makeNullLiteral(TypeBuilder.makeString(nullable));
+            } else {
+                return makeStringLiteral((String) obj);
```
Comment (zzcclp) [inline jvm/src/main/java/io/glutenproject/substrait/expression/ExpressionBuilder.java:116]: Use `obj.toString()` here, when obj is StringType, in fact it's `UTF8String`, can not use `(String) obj` to convert UTF8String to String.

**Comment by rui-mo** [comment]
Verified TPC-H on Jenkins.

**Comment by rui-mo** [comment]
Verified TPC-H on Jenkins.

**Comment by zzcclp** [comment]
LGTM, verified with ClickHouse Backend.

**Pull request merged**